### PR TITLE
Mining skill

### PR DESCRIPTION
**Pull request by Promises**
Has: pickaxes, ores and prospecting
lacks: gems and rune essence

changes to cache: added the sound effect 3600 from osrs which plays when an ore is depleted, the same sound effect probably already exists but i was unable to find it

### CODE REVIEW

File: src/plugins/skills/mining/mining.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
+import {ActionType, RunePlugin} from "@server/plugins/plugin";
```
Comment (TheBlackParade) [inline src/plugins/skills/mining/mining.ts:1]: Please add spaces within import braces and use single ticks. `import { ActionType, RunePlugin } from '...';`

File: src/plugins/skills/mining/mining.ts [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+import {cache, world} from "@server/game-server";
+import {soundIds} from "@server/world/config/sound-ids";
+
+const action: objectAction = (details) => {
+    // Get the mining details for the target rock
+    const ore = getOreFromRock(details.object.objectId);
+
+    // getOreFromRock only returns ores that are minable
+    if (!ore) {
+        details.player.sendMessage("There is current no ore available in this rock.")
```
Comment (TheBlackParade) [inline src/plugins/skills/mining/mining.ts:17]: Single or backticks preferred

File: src/plugins/skills/mining/prospecting.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import {ActionType, RunePlugin} from "@server/plugins/plugin";
```
Comment (TheBlackParade) [inline src/plugins/skills/mining/prospecting.ts:1]: Don't forget these guys down here :)

File: src/plugins/skills/mining/prospecting.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+const action: objectAction = (details) => {
+    details.player.sendMessage("You examine the rock for ores.")
+    details.player.face(details.position);
+    const ore = getOreFromRock(details.object.objectId);
+    details.player.playSound(soundIds.oreEmpty, 7, 0);
+
+    setTimeout(() => {
+        if(!ore) {
+            details.player.sendMessage("There is current no ore available in this rock.")
```
Comment (TheBlackParade) [inline src/plugins/skills/mining/prospecting.ts:16]: There is `currently` no ore available

File: src/world/config/pickaxe.ts [hunk 5]
```diff
@@ -0,0 +1,49 @@
+import {Player} from "@server/world/actor/player/player";
```
Comment (TheBlackParade) [inline src/world/config/pickaxe.ts:1]: These imports as well~

**Comment by TheBlackParade** [comment]
I think the sound effect you want is 2659 or 2661 ?? for the ore depletion

**Comment by Promises** [comment]
@TheBlackParade no, that one is for ores that are already empty, this one is when it becomes empty :/

**Comment by TheBlackParade** [comment]
Ahhh, damn.

**Pull request merged**